packages/smithy-http/src/smithy_http/serializers.py

```diff
@@ -112,6 +112,10 @@ def begin_struct(self, schema: Schema) -> Iterator[ShapeSerializer]:
         ) is not None and not iscoroutinefunction(seek):
             seek(0)
 
+        # TODO: conditional on empty-ness and a param of the protocol?
+        headers = binding_serializer.header_serializer.headers
+        headers.append(("content-type", self._payload_codec.media_type))
```

Comment:
If there is a payload member, it'll be the content-type of said member (`application/octet-stream`, `text/plain`, `MediaType` trait value, event stream content type, or codec content type). If there's no payload member, it's just the codec type.

The emptyness thing can be done later for sure. We might have to add some tracking to see if anything was serialized since inspecting the payload isn't a good idea.